### DISCUSSION TITLE
有关报文的问题

### DISCUSSION THREAD

**Discussion by novada0** [SUSTech-CS305-Fall22/CS305-Project-Skeleton#31] [Q&A]
1、请问报文的payload长度是1024byte吗？代码上是这样写的，文档上貌似没说。
2、如果我想发送一个WHOHAS的报文，可以把所以要请求的datahash连续放在payload中吗？example代码好像每次只取了payload的前20个字节，这里我不太明白。
谢谢！

**Comment by novada0** [comment]
补充一下，第一个问题忘打了max，应该是“请问报文的max payload长度是1024byte吗？”

**Comment by TANSixu** [comment]
Hi, payload最长1400，不超过1400的任何长度都可以。

请求的datahash都应该放在WHOHAS的payload中，example只实现了一个chunk的下载，在example的注释中有提到。测试数据的datahash加起来保证不会超过payload限制。

**Reply by Siri-Lee** [reply]
但是描述文档中写的是“The max packet length (header+payload) is 1400Byte”，所以是Payload最长1400还是（payload + header最长）是1400呢？
以及代码框架中给到的“MAX_PAYLOAD = 1024”是否跟上述描述有所冲突呢？

**Reply by TANSixu** [reply]
> 但是描述文档中写的是“The max packet length (header+payload) is 1400Byte”，所以是Payload最长1400还是（payload + header最长）是1400呢？ 以及代码框架中给到的“MAX_PAYLOAD = 1024”是否跟上述描述有所冲突呢？

抱歉，可能忘记了解释，上一个回答里的payload指调用udp sendto时的payload，对于我们的实现来讲，应该是（header+payload（our payload））。MAX_PAYLOAD可以自己定，不超过1400就可以。或者说，1400是MAX_PAYLOAD的最大长度。

**Reply by Siri-Lee** [reply]
好的谢谢！
